IllegalArgumentException (zipcoder, addAvailableThemes)

From Google Play Console and new for our current beta 2021.03.16-RC1:

30 occurrences, 1 user, Android 11, Google Pixel 3a

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3431)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3595)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:85)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2066)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:223)
  at android.app.ActivityThread.main (ActivityThread.java:7660)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:592)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:947)
Caused by: java.lang.IllegalArgumentException: 
  at java.util.zip.ZipCoder.toString (ZipCoder.java:65)
  at java.util.zip.ZipInputStream.readLOC (ZipInputStream.java:318)
  at java.util.zip.ZipInputStream.getNextEntry (ZipInputStream.java:124)
  at org.mapsforge.map.rendertheme.ZipXmlThemeResourceProvider.scanXmlThemes (ZipXmlThemeResourceProvider.java:134)
  at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.addAvailableThemes (RenderThemeHelper.java:409)
  at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.recalculateAvailableThemes (RenderThemeHelper.java:377)
  at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.lambda$resynchronizeOrDeleteMapThemeFolder$1 (RenderThemeHelper.java:368)
  at cgeo.geocaching.maps.mapsforge.v6.-$$Lambda$RenderThemeHelper$-Uzv_49yNPRGKyKE3eGpdd0EkNY.accept (Unknown Source:2)
  at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper$MapThemeFolderSynchronizer.requestResynchronization (RenderThemeHelper.java:463)
  at cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper.resynchronizeOrDeleteMapThemeFolder (RenderThemeHelper.java:366)
  at cgeo.geocaching.MainActivity.onCreate (MainActivity.java:338)
  at android.app.Activity.performCreate (Activity.java:8000)
  at android.app.Activity.performCreate (Activity.java:7984)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1309)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3404)
```

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
@@ -189,11 +189,17 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback, SharedPref
             //always cache the last used ZipResourceProvider. Check if current one can be reused, if not then reload
             synchronized (cachedZipMutex) {
                 if (cachedZipProvider == null || !themeIdTokens[0].equals(cachedZipProviderFilename)) {
-                    //reload
-                    cachedZipProvider = new ZipXmlThemeResourceProvider(new ZipInputStream(ContentStorage.get().openForRead(theme.fileInfo.uri)), ZIP_RESOURCE_READ_LIMIT);
-                    cachedZipProviderFilename = themeIdTokens[0];
+                    try {
+                        //reload
+                        cachedZipProvider = new ZipXmlThemeResourceProvider(new ZipInputStream(ContentStorage.get().openForRead(theme.fileInfo.uri)), ZIP_RESOURCE_READ_LIMIT);
+                        cachedZipProviderFilename = themeIdTokens[0];
+                    } catch (Exception ex) {
+                        Log.w("Problem loading ZIP Theme '" + theme.fileInfo.uri + "'", ex);
+                        cachedZipProvider = null;
+                        cachedZipProviderFilename = null;
+                    }
                 }
-                xmlRenderTheme = new ZipRenderTheme(themeIdTokens[1], cachedZipProvider, this);
+                xmlRenderTheme = cachedZipProvider == null ? null : new ZipRenderTheme(themeIdTokens[1], cachedZipProvider, this);
             }
         }
         return xmlRenderTheme;
@@ -412,6 +418,8 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback, SharedPref
                     }
                 } catch (IOException ioe) {
                     Log.w("Map Theme ZIP '" + candidate + "' could not be read", ioe);
+                } catch (Exception e) {
+                    Log.w("Problem opening map Theme ZIP '" + candidate + "'", e);
                 }
             }
         }

